### ISSUE TITLE
Uncaught error when given date beyond max date

### ISSUE THREAD

**Issue by jdanil** [wojtekmaj/react-date-picker#69]
When entering a year beyond the max date, react-date-picker crashes the app.
This is reproducible on the [demo](http://projects.wojtekmaj.pl/react-date-picker/) page.

![image](https://user-images.githubusercontent.com/8342105/39412529-0818764a-4c61-11e8-9b55-0d7c49ad4016.png)

To reproduce, enter any date beyond the max date in JS (i.e. Sat Sep 13 275760). So most 6 and any 7 digit numbers will crash the app.

It evaluates to an "Invalid Date" which is then passed to `getProcessedValue`.

Not sure what fix would be appropriate. Maybe adding an additional check to `onChangeExternal` before getting the `processedValue`?
```
const proposedValue = new Date(values.year, (values.month || 1) - 1, values.day || 1);
if (!isNaN(proposedValue) {
        const processedValue = this.getProcessedValue(proposedValue);
        this.props.onChange(processedValue, false);
}
```

**Comment by jdanil** [comment]
@wojtekmaj, happy to contribute if you are open to PRs. Is the proposal above alright, or do you have another fix in mind?

**Comment by wojtekmaj** [comment]
Hey, sorry for the wait. Super busy lately. Actually, a line before `const proposedValue` is supposed to check for validity:
`if (formElements.every(formElement => formElement.value && formElement.checkValidity()))`
and it looks to me like year input is not properly validated. So the fix should be probably in YearInput component, which I thought is pretty straightforward, but I don't know yet why this happens. Will need to investigate.

**Comment by wojtekmaj** [comment]
Fixed in the v6.10.4. Thanks!
